framework/wazuh/agent.py

```diff
@@ -676,55 +679,107 @@ def get_outdated_agents(agent_list=None, offset=0, limit=common.database_limit,
+            if agent_result['error'] == 0:
+                task_agent = {
+                    'agent': str(agent_result['agent']).zfill(3),
+                    'task_id': agent_result['task_id']
+                }
+                result.affected_items.append(task_agent)
+                result.total_affected_items += 1
+            else:
+                error = WazuhError(code=1810 + agent_result['error'], cmd_error=True,
+                                   extra_message=agent_result['message'])
+                result.add_failed_item(id_=str(agent_result['agent']).zfill(3), error=error)
+    result.affected_items = sorted(result.affected_items, key=lambda k: k['agent'])
 
-    return Agent(agent_id).upgrade_result(timeout=int(timeout))
+    return result
 
 
-@expose_resources(actions=["agent:upgrade"], resources=["agent:id:{agent_list}"], post_proc_func=None)
-def upgrade_agents_custom(agent_list=None, file_path=None, installer=None):
+@expose_resources(actions=["agent:upgrade"], resources=["agent:id:{agent_list}"],
+                  post_proc_kwargs={'exclude_codes': [1703, 1817]})
+def get_upgrade_result(agent_list=None):
     """Read upgrade result output from agent.
 
-    :param agent_list: List of agents ID's.
-    :param file_path: Path to the installation file.
-    :param installer: Selected installer.
-    :return: Upgrade message.
+    Parameters
+    ----------
+    agent_list : list
+        List of agent ID's.
+
+    Returns
+    -------
+    Upgrade result.
     """
-    # We access unique agent_id from list, this may change if and when we decide to add option to upgrade a list of
-    # agents
-    agent_id = agent_list[0]
+    result = AffectedItemsWazuhResult(all_msg='All agents have been updated',
+                                      some_msg='Some agents have not been updated',
+                                      none_msg='No agent has been updated')
+
+    agent_list = list(map(int, agents_padding(result=result, agent_list=agent_list)))
+    agents_result_chunks = [agent_list[x:x + 100] for x in range(0, len(agent_list), 100)]
+
+    task_results = list()
+    for agents_chunk in agents_result_chunks:
+        task_results.append(core_upgrade_agents(agents_chunk=agents_chunk, get_result=True))
+
+    for task_result_chunk in task_results:
+        for task_result in task_result_chunk['data']:
+            task_error = task_result.pop('error')
+            if task_error == 0:
+                task_result['agent'] = str(task_result['agent']).zfill(3)
+                result.affected_items.append(task_result)
+                result.total_affected_items += 1
+            else:
+                error = WazuhError(code=1810 + task_error, cmd_error=True, extra_message=task_result['message'])
+                result.add_failed_item(id_=str(task_result.pop('agent')).zfill(3), error=error)
+    result.affected_items = sorted(result.affected_items, key=lambda k: k['agent'])
```

Comment:
I think this is not the proper way to sort a distributed call

Reply:
Updated, now it is sorted when we receive the whole result from the api distributed call